spec/mongo/bulk_write_spec.rb

```diff
@@ -425,4 +433,44 @@
       end
     end
   end
+
+  describe '#bypass_document_validation', if: find_command_enabled? do
+
+    context 'when the documents are invalid' do
+
+      before do
+        collection_with_validator.insert_many([{:a => 1}, {:a => 2}])
+      end
+
+      let(:requests) do
+        [
+            { insert_one: { :x => 1 } },
+            { update_one: { filter: { :a => 1 },
+                            update: { '$unset' => { :a => '' }} } },
+            { replace_one: { filter: { :a => 2 },
+                             replacement: { :x => 2 }}}
+        ]
+      end
+
+      let(:result) do
+        collection_with_validator.bulk_write(requests)
+      end
+
+      it 'raises BulkWriteError' do
+        expect {
+          result
+        }.to raise_exception(Mongo::Error::BulkWriteError)
+      end
+
+      let(:result2) do
+        collection_with_validator.bulk_write(
+            requests, :bypass_document_validation => true)
+      end
```

Comment:
actually, this makes so much more sense as you can see the result right before the test. You can leave it like this, thanks = )